assets/app/scripts/extensions/javaLink.js

```diff
@@ -40,25 +40,27 @@ angular.module('openshiftConsoleExtensions', ['openshiftConsole'])
       if (!jolokiaPort) {
         return;
       }
-      var pod = $scope.$eval('pod');
+      var pod = $scope.$eval('podTemplate');
       if (!pod) {
         return;
       }
       var podId = pod.metadata.name;
       var namespace = pod.metadata.namespace;
-      $scope.jolokiaUrl = new URI(ProxyPod(namespace, podId, jolokiaPort.containerPort)).segment('jolokia/').toString();
+      $scope.jolokiaUrl = ProxyPod(namespace, podId, jolokiaPort.containerPort).segment('jolokia/').toString();
       $scope.gotoContainerView = function($event, container, jolokiaUrl) {
         $event.preventDefault();
         $event.stopPropagation();
         var returnTo = window.location.href;
         var title = container.name || 'Untitled Container';
+        var token = LocalStorageUserStore.getToken();
         var targetURI = new URI().path(BaseHref)
                                  .segment('java')
                                  .segment('index.html')
                                  .query({
                                    jolokiaUrl: jolokiaUrl,
                                    title: title,
-                                   returnTo: returnTo
+                                   returnTo: returnTo,
+                                   token: token
```

Comment:
1. Switch to putting the token in the hash so it never gets sent to the server
2. Immediately rewrite the URL to remove it (sounds like that's happening already), but make sure you don't break the back button... clicking back should bring you back, not trap you in a token-removing redirect loop